main.py

```
@@ -0,0 +1,375 @@
+    except httpx.RequestError as e:
+        logger.exception("CMS request error for %s", path, exc_info=e)
+        raise HTTPException(
+            status_code=502, detail="Upstream CMS request failed") from e
+
+    try:
+        return response.json(), response.status_code
+    except ValueError as e:
+        logger.exception(
+            "CMS returned non-JSON response for %s", path, exc_info=e)
+        raise HTTPException(
+            status_code=502, detail="Upstream CMS invalid JSON")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)